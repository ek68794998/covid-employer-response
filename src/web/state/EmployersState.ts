import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}