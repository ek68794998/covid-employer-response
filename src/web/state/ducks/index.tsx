import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});