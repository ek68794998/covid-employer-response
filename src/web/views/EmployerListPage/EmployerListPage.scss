@import "../Colors.scss";
@import "../Dimensions.scss";

$division-padding: 16px;
$top-button-size: 48px;
$top-button-size-small: 32px;

main#employer-list {
	margin: 0 auto;
	width: 100%;

	> div {
		margin: 0 auto;
		padding: 0 $page-padding;
	}

	div.EmployerListPage__Filters {
		box-shadow: 0 0 8px $box-shadow-color;
		background: $background;
		padding: $division-padding;
		position: sticky;
		top: 0;
		z-index: 10;
		
		> div {
			margin: 0 auto;
			max-width: ($body-max-width - $page-padding * 2);
			min-width: ($body-min-width - $division-padding * 2);
			width: 50%;
		}
	}
	
	div.EmployerListPage__Content {
		max-width: $body-max-width;
		padding-bottom: $division-padding;
		padding-top: $division-padding;
	}
}

button.EmployerListPage__BackToTop {
	background: $background;
	border: 0;
	border-radius: ($top-button-size / 2);
	bottom: ($page-padding / 2);
	box-shadow: 0 2px 6px $box-shadow-color;
	cursor: pointer;
	position: fixed;
	height: $top-button-size;
	outline: none;
	pointer-events: initial;
	right: ($page-padding / 2);
	width: $top-button-size;
	z-index: 20;

	i {
		font-size: 36px;
		margin-top: 2px;
	}
}

@media (max-width: $body-narrow-width) {
	main#home {
		> div {
			padding: 0 $page-padding-narrow;
		}

		div.EmployerListPage__Filters > div {
			max-width: $body-max-width;
		}
	}

	button.EmployerListPage__BackToTop {
		bottom: ($page-padding-narrow / 2);
		height: $top-button-size-small;
		right: ($page-padding-narrow / 2);
		width: $top-button-size-small;

		i {
			font-size: 24px;
			margin-left: -2px;
		}
	}
}
