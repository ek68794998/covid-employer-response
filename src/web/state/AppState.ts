import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}