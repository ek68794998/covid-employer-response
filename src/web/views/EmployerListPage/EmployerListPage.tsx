import React, { useState } from "react";
import { Helmet } from "react-helmet";
import { useSelector } from "react-redux";
import { withRouter } from "react-router-dom";

import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { getStrings } from "../../state/ducks/localization/selectors";

import BackToTopButton from "../BackToTopButton/BackToTopButton";
import EmployerList from "../EmployerList/EmployerList";
import EmployerListSearch from "../EmployerListSearch/EmployerListSearch";
import { EmployerListSearchFilter } from "../EmployerListSearch/EmployerListSearchFilter";

import "./EmployerListPage.scss";

const EmployerListPage: React.FC = (): React.ReactElement => {
	const [ searchFilters, setSearchFilters ] = useState(new EmployerListSearchFilter());

	const strings: LocalizedStrings = useSelector(getStrings);

	return (
		<main id="employer-list">
			<Helmet>
				<title>{strings.employerList} | {strings.appTitle}</title>
			</Helmet>
			<div className="EmployerListPage__Filters">
				<EmployerListSearch onChange={setSearchFilters} />
			</div>
			<div className="EmployerListPage__Content">
				<EmployerList searchFilter={searchFilters} />
			</div>
			<BackToTopButton />
		</main>
	);
};

export default withRouter(EmployerListPage);
