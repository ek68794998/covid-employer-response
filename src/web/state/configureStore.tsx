import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;