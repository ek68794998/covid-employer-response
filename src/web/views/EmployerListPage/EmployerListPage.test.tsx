import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { getPlocStringsAsync, mockComponent } from "../../../__tests__/TestUtils";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerListPage from "./EmployerListPage";

jest.mock(
	"../EmployerListSearch/EmployerListSearch",
	() => mockComponent("EmployerListSearch"));

jest.mock(
	"../EmployerList/EmployerList",
	() => mockComponent("EmployerList"));

describe("<EmployerListPage />", () => {
	test("renders without exploding", async () => {
		const store: Store<AppState, AnyAction> = configureStore({ strings: await getPlocStringsAsync() });

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerListPage />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
