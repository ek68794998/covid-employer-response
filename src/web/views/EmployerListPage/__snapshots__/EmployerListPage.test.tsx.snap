// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerListPage /> renders without exploding 1`] = `
<main
  id="employer-list"
>
  <div
    className="EmployerListPage__Filters"
  >
    [component: EmployerListSearch /]
  </div>
  <div
    className="EmployerListPage__Container"
  >
    <div
      className="EmployerListPage__Content"
    >
      [component: EmployerList /]
    </div>
  </div>
</main>
`;
